src/Workspaces/CSharp/Portable/Serialization/CSharpOptionsSerializationService.cs

```diff
@@ -28,7 +29,7 @@ public override void WriteTo(CompilationOptions options, ObjectWriter writer, Ca
             WriteCompilationOptionsTo(options, writer, cancellationToken);
 
             var csharpOptions = (CSharpCompilationOptions)options;
-            writer.WriteValue(csharpOptions.Usings.ToArray());
+            writer.WriteArray(csharpOptions.Usings, static (w, u) => w.WriteString(u));
```

Comment:
arrays go through the helper path *external* to the writer itself. t his is just an extension that writes out the count, then each item one at a time.